### PR TITLE
Implements the Kafka driver

### PR DESCRIPTION
**Pull request by sileht**

### CODE REVIEW

File: pifpaf/drivers/kafka.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+log.segment.bytes=1073741824
+log.retention.check.interval.ms=300000
+zookeeper.connect=localhost:%d
+zookeeper.connection.timeout.ms=6000
+""" % (self.port, self.port, logdir, self.zookeeper_port))
+
+        # NOTE(sileht): The wait_for_line is the best we can do
+        # but we error can occur after the last line we see when it works...
+        env = {"LOG_DIR": logdir}
+        self._exec(['zookeeper-server-start.sh', zookeeper_conf],
```
Comment (jd) [inline pifpaf/drivers/kafka.py]: Is Kafka shipping ZK or is there any other reason not to use ZK from pifpaf? :(

File: pifpaf/drivers/__init__.py [hunk 2]
```diff
@@ -148,13 +150,32 @@ def _exec(self, command, stdout=False, ignore_failure=False,
...
                 decoded_line = fsdecode(line)
+
                 if wait_for_line and re.search(wait_for_line,
                                                decoded_line):
                     break
             stdout_str = b"".join(lines)
         else:
             stdout_str = None
 
+        if (stdout or wait_for_line) and forbidden_line_after_start:
```
Comment (jd) [inline pifpaf/drivers/__init__.py]: glups :/

damn software

File: pifpaf/drivers/kafka.py (hunk 1 above)
Comment (sileht) [inline pifpaf/drivers/kafka.py]: yeah, kafka ships it own preconfigured version of zookeeper.

Comment (sileht) [inline pifpaf/drivers/kafka.py]: When distribution will ship kafka I think we have to rewrite that, but for now...

**Comment by jd** [comment]
This does not work on OS X using Homebrew, as it misses `/usr/local/opt/kafka/bin` in `PATH`  to access the `zookeeper-*` stuff.

**Comment by jd** [comment]
@sileht fun fact, it's actually `/usr/local/opt/kafka/bin/zookeeper-server-start` without `.sh` on Homebrew :( Sorry I missed that too.

**Comment by harlowja** [comment]
Is this based on the other PR, me confused, ha.

**Comment by sileht** [comment]
yes, I dunno how to keep the PR number. But I have keep the orignal author :)

**Comment by jd** [comment]
`TypeError: find_executable() takes exactly 2 arguments (1 given)`

**Comment by sileht** [comment]
it's fixed.

**Pull request merged**